datatypes/src/primitives/time_step.rs

```diff
@@ -438,65 +390,73 @@ impl Sub<TimeStep> for TimeInterval {
     }
 }
 
+impl Mul<TimeStep> for u32 {
+    type Output = TimeStep;
+
+    fn mul(self, rhs: TimeStep) -> Self::Output {
+        TimeStep {
+            granularity: rhs.granularity,
+            step: self * rhs.step,
+        }
+    }
+}
+
 /// An `Iterator` to iterate over time in steps
 #[derive(Debug, Clone)]
 pub struct TimeStepIter {
     reference_time: TimeInstance,
     time_step: TimeStep,
     curr: u32,
-    max: u32,
+    steps: u32,
 }
 
 impl TimeStepIter {
-    /// Create a new `TimeStepIter` which will include the start `TimeInstance`.
+    /// Create a new `TimeStepIter` with given amount of `steps`.
     /// # Errors
     /// This method fails if the interval [start, max) is not valid in chrono.
-    pub fn new_incl_start(
-        reference_time: TimeInstance,
-        time_step: TimeStep,
-        steps: u32,
-    ) -> Result<Self> {
+    pub fn new(reference_time: TimeInstance, time_step: TimeStep, steps: u32) -> Result<Self> {
+        ensure!(
```

Comment:
Yes. We had the issue that you couldn't shift the default (eternal) time because then the endpoints got invalid. I solved this by making any arithmetic operation on the begin and end of time a no-op. Thus, if either start or end of an interval are at begin/end of time, they are anchored there and the other end would get shifted normally. However, this also means we can't use it here as the start of an iterator because it would get stuck. But imho, there is no case where we want an iterator starting at the begin of time. An even more elegant solution might be to make TimeInstance an enum { Begin, Time(u64), End } to make this more verbose, but I think this is to expensive.